### PR TITLE
107729/assign project

### PR DESCRIPTION
**Pull request by jordan-beard**
UI changes for assigning a project. Will look to add tests next.

### CODE REVIEW

File: Frontend.Integration.Tests/Pages/Projects/ProjectAssignment/ProjectAssignmentTests.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		{
+			var project = GetProject();
+			_factory.AddAnyPatch($"/academyTransferProject/{project.ProjectUrn}", project);
+
+			await OpenUrlAsync($"/project-assignment/{project.ProjectUrn}");
+
+			Document.QuerySelector<IHtmlInputElement>("#UnassignDeliveryOfficer").Value = "true";
+			await Document.QuerySelector<IHtmlFormElement>("form").SubmitAsync();
+
+			Document.Url.Should().EndWith($"project/{project.ProjectUrn}");
```
Comment (dneed-nimble) [inline Frontend.Integration.Tests/Pages/Projects/ProjectAssignment/ProjectAssignmentTests.cs:32]: Might be worth having an assertion that the field reads as Empty. To confirm it's not just moved pages but updated the AssignedUser value on the UI.

File: Frontend.Integration.Tests/Pages/Projects/ProjectAssignment/ProjectAssignmentTests.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+			_factory.AddAnyPatch($"/academyTransferProject/{project.ProjectUrn}", project);
+
+			await OpenUrlAsync($"/project-assignment/{project.ProjectUrn}");
+
+			var fullName = "Bob 1";
+
+			Document.QuerySelector<IHtmlOptionElement>($"[value='{fullName}']").IsSelected = true;
+			await Document.QuerySelector<IHtmlFormElement>("form").SubmitAsync();
+
+			Document.Url.Should().EndWith($"project/{project.ProjectUrn}");
```
Comment (dneed-nimble) [inline Frontend.Integration.Tests/Pages/Projects/ProjectAssignment/ProjectAssignmentTests.cs:48]: Might be worth having an assertion that the field reads as Bob 1. To confirm it's not just moved pages but updated the AssignedUser value on the UI.

File: Frontend.Tests/PagesTests/Projects/ProjectAssignment/ProjectAssignmentTests.cs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+				_projectRepository.Setup(m => m.GetByUrn(It.IsAny<string>())).ReturnsAsync(expectedResult);
+
+				var users = _fixture.Create<List<User>>();
+				_userRepository.Setup(m => m.GetAllUsers()).ReturnsAsync(users);
+
+				await _subject.OnPostAsync("12345", users.First().FullName);
+
+				var expectedUser = new User(users.First().Id, users.First().EmailAddress, users.First().FullName);
+
+				_projectRepository.Verify(m => m.Update(It.Is<Project>(p => p.AssignedUser.Should().BeEquivalentTo(expectedUser, "") != null)),
```
[the remark this answers is not in the document]
Comment (jordan-beard) [reply Frontend.Tests/PagesTests/Projects/ProjectAssignment/ProjectAssignmentTests.cs:97]: Its not. I borrowed this, to make an assertion within a verify lambda. 
image

But tbh its not that readable I could change it to check each field individually would that be better?

[the remark this answers is not in the document]
Comment (dneed-nimble) [reply Frontend.Tests/PagesTests/Projects/ProjectAssignment/ProjectAssignmentTests.cs:97]: I think that could be slightly better but now I understand the above I'm happy to approve.

[the remark this answers is not in the document]
Comment (jordan-beard) [reply Frontend.Tests/PagesTests/Projects/ProjectAssignment/ProjectAssignmentTests.cs:97]: I dont mind updating if it makes it clearer.

[the remark this answers is not in the document]
Comment (jordan-beard) [reply Frontend.Tests/PagesTests/Projects/ProjectAssignment/ProjectAssignmentTests.cs:97]: Done.

**Review by dneed-nimble — APPROVED** [review]

**Pull request merged**